src/layout.jl

```diff
@@ -9,10 +9,18 @@ predictor(ifd::IFD) = Int(getdata(ifd, PREDICTOR, 0))
 bitspersample(ifd::IFD) = Int(first(ifd[BITSPERSAMPLE].data))::Int
 ispalette(ifd::IFD) = Int(getdata(ifd, PHOTOMETRIC, 0)) == 3
 compression(ifd::IFD) = getdata(CompressionType, ifd, COMPRESSION, COMPRESSION_NONE)
+colortype(ifd::IFD) =  first(interpretation(ifd)){_mappedtype(rawtype(ifd), bitspersample(ifd))}
 
 is_irregular_bps(ifd::IFD) = bitspersample(ifd) != sizeof(rawtype(ifd)) * 8
 is_complicated(ifd::IFD) = !iscontiguous(ifd) || compression(ifd) != COMPRESSION_NONE || is_irregular_bps(ifd) == true || predictor(ifd) > 1
 
+# returns true if all slices have the same size and color type
+function is_homogeneous(ifds::Vector{<:IFD})
+    return all(map(==(nrows(first(ifds))), nrows.(ifds))) &&
+        all(map(==(ncols(first(ifds))), ncols.(ifds))) &&
+        all(map(==(colortype(first(ifds))), colortype.(ifds)))
+end
+
```

Comment:
Is this sufficient to determine homogeneity? I guess we can try with this and add more checks if they end up causing issues.